server/services/canvas/__init__.py

```diff
@@ -32,25 +32,43 @@ def get_user(canvas_id, key=None) -> FakeUser | User:
 def get_course(canvas_id, key=None) -> FakeCourse | Course:
     return _get_client(key).get_course(canvas_id)
 
+def is_staff_enrollment(enrollment_type: str):
+    return enrollment_type.lower() in ('ta', 'teacher')
 
 def is_course_valid(c) -> bool:
+    # A valid course has a name and id
+    # TBD if we should filter on published
     return not (not c) and \
         hasattr(c, 'id') and \
         hasattr(c, 'name') and \
-        hasattr(c, 'course_code') and \
-        hasattr(c, 'start_at') and \
-        hasattr(c, 'start_at_date')
+        hasattr(c, 'course_code')
+
+def noramlize_course_start_date(course: FakeCourse | Course) -> None:
+    # Ensure a valid start_at date for a course.
+    # return the term start_at_date if present
+    if course['term'] and course['term']['start_at']:
+        start_at = course['term']['start_at']
+        start_at_date = course['term']['start_at_date']
+    else:
+        start_at_date = course['start_at_date'] or course['created_at_date']
+        start_at = course['start_at'] or course['created_at']
+    course['start_at_date'] = start_at_date
+    course['start_at'] = start_at
```

Comment:
wait shouldn't we use dot notation 
`course.start_at` instead `course['start_at']`

Reply:
oh sure...I wasn't sure, I thought these were dicts, but yeah maybe we should... I assume a.b.c is valid syntax then? I don't remember all the rules of Python or if we need to check for presence before an error is realized..

Reply:
No worries I will take it from here.